Content.Shared/Wires/SharedWiresSystem.cs

```diff
@@ -33,4 +75,34 @@ private void OnExamine(EntityUid uid, WiresPanelComponent component, ExaminedEve
             }
         }
     }
+
+    protected void UpdateAppearance(EntityUid uid, WiresPanelComponent panel)
+    {
+        if (TryComp<AppearanceComponent>(uid, out var appearance))
+            Appearance.SetData(uid, WiresVisuals.MaintenancePanelState, panel.Open && panel.Visible, appearance);
+    }
+
+    public void TogglePanel(EntityUid uid, WiresPanelComponent component, bool open)
+    {
+        component.Open = open;
+        UpdateAppearance(uid, component);
+        Dirty(uid, component);
+
+        var ev = new PanelChangedEvent(component.Open);
+        RaiseLocalEvent(uid, ref ev);
+    }
+
+    public bool IsPanelOpen(Entity<WiresPanelComponent?> entity)
+    {
+        if (!Resolve(entity, ref entity.Comp, false))
+            return true;
+
+        // Listen, i don't know what the fuck this component does. it's stapled on shit for airlocks
```

Comment:
Isn't this used for the steel plates inside high-security airlocks that you need to remove before you can cut wires?

Reply:
i acknowledge that that's what the intended use is, but the fact that it just barely integrates into panels is really a testament to the absurd coupling between the two. How it's meant to interact with panels as a whole is honestly up in the air.